core/src/wallet/common/AbstractWallet.cpp

```diff
@@ -48,11 +48,10 @@ namespace ledger {
                                        const api::Currency &currency,
                                        const std::shared_ptr<WalletPool> &pool,
                                        const std::shared_ptr<DynamicObject> &configuration,
-                                       const DerivationScheme &derivationScheme
-        )
-                : DedicatedContext(
-                pool->getDispatcher()->getSerialExecutionContext(fmt::format("wallet_{}", walletName))),
-                  _scheme(derivationScheme) {
+                                       const DerivationScheme &derivationScheme)
+                : DedicatedContext(pool->getDispatcher()->getThreadPoolExecutionContext(fmt::format("wallet_{}", walletName))),
```

Comment:
Do we really need a new EC? Why not using the main thread pool EC and let it dispatch parallelism for us?

Reply:
Nice catch, I forgot to remove this one !